types/react/index.d.ts

```diff
@@ -331,6 +337,14 @@ declare namespace React {
         displayName?: string;
     }
 
+    interface RefForwardingComponent<T, P = {}> {
```

Comment:
Is it actually a component, or just a callback?

Though arguably any function that returns `ReactNode` is a component if it reads its parameters in a compatible way.

Reply:
The source code even calls it `render` as its parameter name, I called it a component here to make it look and feel more familiar to userland, as I think a lot of use cases for this would be inlining a functional component like:

```tsx
const MyComponent = React.forwardRef(
  function MyComponent(props: {}, ref: React.Ref<HTMLDivElement>) {
    return <div ref={ref} />
})
```

and having the interface have the word `Component` in it will give a bit of extra guidance as to how the user should structure the callback.